Test/Unit/Helper/CartTest.php

```diff
@@ -0,0 +1,433 @@
+            ->willReturn($billingAddress);
+        $quote->method('getShippingAddress')
+            ->willReturn($shippingAddress);
+        $quote->method('getQuoteCurrencyCode')
+            ->willReturn('$');
+        $quote->method('collectTotals')
+            ->willReturnSelf();
+        $quote->method('getTotals')
+            ->willReturn([]);
+        $quote->method('getStoreId')
+            ->willReturn(1);
+        $quote->method('getUseRewardPoints')
+            ->willReturn(false);
+        $quote->method('getUseCustomerBalance')
+            ->willReturn(false);
+
+        return $quote;
+    }
+
+    /**
+     * @param bool $shouldBeEmpty
+     * @return \PHPUnit_Framework_MockObject_MockObject
+     */
+    private function getBillingAddress($shouldBeEmpty = false)
+    {
+        $addressData = $this->getAddressData($shouldBeEmpty);
+        $billingAddress = $this->getMockBuilder(Quote\Address::class)
+            ->setMethods([
+                'getFirstname', 'getLastname', 'getCompany', 'getTelephone', 'getStreetLine',
+                'getCity', 'getRegion', 'getPostcode', 'getCountryId', 'getEmail',
+                'getDiscountAmount'
+            ])
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $billingAddress->method('getFirstname')
+            ->willReturn($addressData['first_name']);
+        $billingAddress->method('getLastname')
+            ->willReturn($addressData['last_name']);
+        $billingAddress->method('getCompany')
+            ->willReturn($addressData['company']);
+        $billingAddress->method('getTelephone')
+            ->willReturn($addressData['phone']);
+        $billingAddress->method('getStreetLine')
+            ->will($this->returnValueMap([
+                [1, $addressData['street_address1']],
+                [2, $addressData['street_address2']]
+            ]));
+        $billingAddress->method('getCity')
+            ->willReturn($addressData['locality']);
+        $billingAddress->method('getRegion')
+            ->willReturn($addressData['region']);
+        $billingAddress->method('getPostcode')
+            ->willReturn($addressData['postal_code']);
+        $billingAddress->method('getCountryId')
+            ->willReturn($addressData['country_code']);
+        $billingAddress->method('getEmail')
+            ->willReturn($addressData['email']);
+        $billingAddress->method('getDiscountAmount')
+            ->willReturn(0);
+
+        return $billingAddress;
+    }
+
+    /**
+     * @param bool $shouldBeEmpty
+     * @return \PHPUnit_Framework_MockObject_MockObject
+     */
+    private function getShippingAddress($shouldBeEmpty = false)
+    {
+        return $this->getBillingAddress($shouldBeEmpty);
+    }
+
+    /**
+     * @param bool $shouldBeEmpty
+     * @return array
+     */
+    private function getAddressData($shouldBeEmpty = false)
+    {
+        if ($shouldBeEmpty) {
+            $data = [
```

Comment:
Please add doc string explaining shouldBeEmpty, it is confusing returning non empty data.

Reply:
Ah yeah, it is extra parameters. I will remove it.